mex.go

```diff
@@ -228,6 +237,17 @@ func (mex *messageExchange) recvPeerFrameOfType(msgType messageType) (*Frame, er
 	}
 }
 
+func (mex *messageExchange) onCtxErr(err error) {
+	// On canceled contexts, we may need to send a cancel message.
+	if err != context.Canceled {
+		return
+	}
+
+	if onCancel := mex.mexset.onCancel; onCancel != nil {
```

Comment:
nit: I think this condition is unnecessary - unless I'm misreading, we're unconditionally setting `onCancel` on the outbound exchange in `connection.go`, it's just the body of `onCancel` that is conditional